jax/interpreters/sharded_jit.py

```diff
@@ -144,6 +144,35 @@ def _partitioned_sharding_spec(num_partitions: int,
         replication_factor=1)
 
 
+def _sharded_jit_translation_rule(c, axis_env, in_nodes, name_stack,
+                                  in_parts, out_parts_thunk, num_partitions,
+                                  backend, name, call_jaxpr):
+  subc = xc.XlaBuilder(f"sharded_jit_{name}")
+
+  # We assume any extra leading in_nodes are constants and replicate them.
+  num_extra_nodes = len(in_nodes) - len(in_parts)
+  assert num_extra_nodes >= 0
+  in_parts = (None,) * num_extra_nodes + in_parts
+
+  args = []
+  for i, (n, sharding) in enumerate(safe_zip(in_nodes, in_parts)):
+    # We use xb.set_sharding instead of xb.with_sharding because inlined calls
+    # shouldn't have shardings set directly on the inputs or outputs.
+    arg = xb.parameter(subc, i, c.GetShape(n))
+    args.append(xb.set_sharding(subc, arg, sharding))
+
+  out_nodes = xla.jaxpr_subcomp(
+      subc, call_jaxpr, backend, axis_env, (),
+      extend_name_stack(name_stack, wrap_name(name, "sharded_jit")), *args)
```

Comment:
True. It doesn't seem very principled or robust to depend on the name strings, but it would work in practice (unless you call a non-pmapped function "pmap" or something, but that's silly). I'm gonna leave it as-is for now, but will keep this in mind as a potential option.